### ISSUE TITLE
[Accessibility]: Improve global announcements

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#34504]
### Audit type

Screen reader navigation

### Problem description

A common problem in our accessibility audits is going to be that we are not announcing information "live" to the user.

For example, see https://github.com/sourcegraph/sourcegraph/issues/34484. This is going to be a continual problem on most use journeys that involve any mutations to back end data.

We should be using [ARIA live regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) to communicate this better.

Related to https://github.com/sourcegraph/sourcegraph/issues/34484

### Expected behavior

This issue is focused on implemented a global solution to these announcements, not specific to a certain user journey.

Right now I see two different variations:

**Visual messages**
Where we already show an `<Alert />` component with some text - like after submitting the [user profile form](https://sourcegraph.com/user/settings/profile).

Solution: Add `role="alert"` to `<Alert />`. This should handle most cases.

**Hidden messages**
Where we do not show anything visible, but we still need the user to be aware that something has happened - like [when a modal closes after adding a credential](https://github.com/sourcegraph/sourcegraph/issues/34484).

Solution: Implement global `screenReaderAnnounce` method. This can work similarly to `eventLog` in that we should just need to fire it in an `onClick|onCompleted` callback.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
